pkg/river/token/builder/token.go

```diff
@@ -0,0 +1,62 @@
+package builder
+
+import (
+	"bytes"
+	"io"
+
+	"github.com/grafana/agent/pkg/river/parser"
+	"github.com/grafana/agent/pkg/river/printer"
+	"github.com/grafana/agent/pkg/river/token"
+)
+
+// LiteralTok is used as a token.Token value for writing literal sequences of
+// text into a built file.
+//
+// river/token doesn't have the capability of representing literal strings, so
+// LiteralTok uses token.ILLEGAL as a way to uniquely identify when literal
+// strings should be injected.
+const LiteralTok = token.ILLEGAL
```

Comment:
It could be any literal text. We could add token.LITERAL, but we'd have to make it clear that the scanner never emits it and it's just for the token builder package.

WDYT?

Reply:
That feels better to me than the verbiage of illegal, but if it makes more sense to you to use illegal then keep it.

Reply:
I think it's probably fine to add a token.LITERAL since there's multiple consumers of the token package anyway